CMakeLists.txt

```diff
@@ -17,9 +17,9 @@ set(CMAKE_CXX_STANDARD_REQUIRED True)
 # Project structure ###########################################################
 #
 # temporary build directories
-set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
-set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
-set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}/lib)
```

Comment:
this should not be changed.

temporary build directories are in the build dir, not in the source (which should stay untouched during build :) )